keps/sig-node/3573-device-plugin/README.md

```diff
@@ -0,0 +1,911 @@
+
+###### What specific metrics should inform a rollback?
+
+`device_plugin_registration_total` and `device_plugin_alloc_duration_seconds` metrics can
+determine the heath of the feature and can be be used to detrmine if a rollback needs to
+be performed.
+
+`device_plugin_registration_total` is an indicator of number of device plugin
+registrations. This metric can be observed both before and after an upgrade. If the
+number of device plugin registrations remain the same, it can be assumed that
+the upgrade process was successful. If however, the number of registrations are not
+same, upgrade process hasn't been successful and a rollback is required.
+
+`device_plugin_alloc_duration_seconds` can be used as an indicator to determine
+whether or not device plugin allocations are taking place successfully. If devices
+were being allocated before upgrade and are not after an update a rollback is
+required.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Yes
+
+When upgrading device plugins, it is recommended that the cluster admin drain workloads
+requesting devices and conform to the device plugin API.
+
+This has been validated by several device plugins. Here is an example: https://github.com/NVIDIA/k8s-device-plugin#upgrading-kubernetes-with-the-device-plugin. 
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+1. The operator can look at `device_plugin_registration_total` and `device_plugin_alloc_duration_seconds`
+   metrics to gather information about successful device plugin registrations and device
+   allocations to pods respectively.
+1. Kubelet PodResource API endpoints (`List` and `GetAllocatableResources`) can be used
+   to obtain information on devices allocated to running pods. Refer to
+   [keps/sig-node/606-compute-device-assignment]([keps/sig-node/606-compute-device-assignment]).
+1. Status of the nodes  specifically device `capacity` and `allocatable` can be inspected
+   to see if device plugins were successfully deployed and are available to be allocated
+   on the node.
+   In addition to that, status of the running pods can be inspected to determine if the
+   pod requesting device(s) was running successfully and allocated device(s) as per its
+   request.
+1. Checking that the pod spec contains request for resources provided by plugins which
+   can be identified as non-native resources (i.e resources other than cpu, memory,
+   hugepages).
+
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] API .status
+  - Field: Node.status
+    - Property: [capacity,allocatable]
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+Not Applicable, refer to Kubelet SLOs.
```

Comment:
`device_plugin_alloc_duration_seconds` seems reasonable to check and decide how long is too long.